File: PYTHON/soal2.py
 # Nama Anggota : 
#  - Fasya Nurina Izzati (140810200052)
#  - Amariel Danendra Dagna (140810200058)
#  - Alya Raisa Hidayat (140810200060)
#  - Muthia Azzahra (140810200066)
# Kelas           : B
# Tanggal Buat    : 1 November 2021
# Deskripsi       : Soal 2 - Python

class Matriks:
    def __init__(self, baris, kolom, nilai):
        self.__baris = baris
        self.__kolom = kolom
        self.__nilai = nilai
    
    def inputBaris(self):
        self.__baris = int(input("Masukan Banyak Baris : "))
    
    def inputKolom(self):
        self.__kolom = int(input("Masukan Banyak Kolom : "))

    def isiMatriks(self):
        for i in range (0, self.__baris):
            kolom = []
            for j in range (0, self.__kolom):
              kolom.append(int(input("Masukan nilai Matriks ke (" + str(i+1) + ", " + str(j+1) + ") = ")))
            self.__nilai.append(kolom)
    
    def printMatriks(self):
        for i in self.__nilai:
            print(i)

    def jumlahBaris(self):
        temp = [0 for i in range(self.__baris)]
        for i in range(self.__baris):
            for j in range (self.__kolom):
                temp [i] += self.__nilai[i][j]
        for i in range(self.__baris):
            print (" * Baris ke-" + str(i+1) + " = " + str(temp[i]))

    def jumlahKolom(self):
        temp = [0 for i in range(self.__kolom)]
        for i in range(self.__baris):
            for j in range (self.__kolom):
                temp [i] += self.__nilai[j][i]
        for i in range(self.__baris):
            print (" * Kolom ke-" + str(i+1) + " = " + str(temp[i]))
    

if __name__ == "__main__":
    baris = 0
    kolom = 0
    mtx = Matriks(baris, kolom, [])
    
    print ("--------------------------------------")
    mtx.inputBaris()
    mtx.inputKolom()

    print ("--------------------------------------")
    mtx.isiMatriks()

    print ("--------------------------------------")
    print ("Bentuk Matriks")
    mtx.printMatriks()

    print ("--------------------------------------")
    print ("Jumlah Baris")
    mtx.jumlahBaris()

    print ("--------------------------------------")
    print ("Jumlah Kolom")
    mtx.jumlahKolom()    

File: CPP/soal2.cpp
#include <iostream>

using namespace std;

class Matriks {
    private:
        int baris;
        int kolom;
        int nilai[10][10];

    public:
        Matriks() {
            this->baris = 10;
            this->kolom = 10;
            nilai[baris][kolom];
        }

        Matriks(int baris, int kolom) {
            this->baris = baris;
            this->kolom = kolom;
            nilai[baris][kolom];
        }

        void isiMatriks() {
            cout << "Masukkan Banyak Baris: "; 
            cin >> this->baris;
            cout << "Masukkan Banyak Kolom: "; 
            cin >> this->kolom;
            for(int i=0; i<this->baris; i++) {
                for(int j=0; j<this->kolom; j++) {
                    cout << "Masukkan nilai Matriks ke (" << i+1 << ", " << j+1 << ") = "; 
                    cin >> this->nilai[i][j];
                }
            }
        }

        void printMatriks() {
            for(int i=0; i<this->baris; i++) {
                for(int j=0; j<this->kolom; j++) {
                    cout << this->nilai[i][j] << "  ";
                }
                cout << endl;
            }
        }

        void jumlahBaris() {
            int temp;
            int hasil[10];

            for(int i=0; i<this->baris; i++) {
                temp = 0;
                for(int j=0; j<this->kolom; j++) {
                    temp += this->nilai[i][j];

                }
                hasil[i] = temp;
                cout << " * Baris ke-" << i+1 << " = " << hasil[i] << endl;
            }
        }

        void jumlahKolom() {
            int temp;
            int hasil[10];

            for(int i=0; i<this->baris; i++) {
                temp = 0;
                for(int j=0; j<this->kolom; j++) {
                    temp += this->nilai[j][i];

                }
                hasil[i] = temp;
                cout << " * Kolom ke-" << i+1 << " = " << hasil[i] << endl;
            }
        }
};

int main() {
    int baris;
    int kolom;

    Matriks *mtx = new Matriks();

    cout << "----------------------------------------" << endl;
    mtx->isiMatriks();
    cout << "----------------------------------------" << endl;
    cout << "Bentuk Matriks" << endl;
    mtx->printMatriks();
    cout << "----------------------------------------" << endl;
    cout << "Jumlah Baris" << endl;
    mtx->jumlahBaris();
    cout << "----------------------------------------" << endl;
    cout << "Jumlah Kolom" << endl;
    mtx->jumlahKolom();
}